Tests/ComposableArchitectureTests/AppStorageTests.swift

```diff
@@ -24,6 +24,33 @@ final class AppStorageTests: XCTestCase {
     XCTAssertEqual(defaults.integer(forKey: "count"), 43)
   }
 
+  func testDefaultsReadURL() {
+    @Dependency(\.defaultAppStorage) var defaults
+    defaults.set(URL(string: "https://pointfree.co"), forKey: "url")
+    @Shared(.appStorage("url")) var url: URL?
+    XCTAssertEqual(url, URL(string: "https://pointfree.co"))
```

Comment:
without the changes, this test would fail because UserDefaults.objectForKey would return Data which is not castable to URL